README.md

```diff
@@ -1,13 +1,17 @@
-# Dal
+# DAL
 
 ![CircleCI](https://img.shields.io/circleci/project/github/expert360/dal.svg)
 ![Codecov](https://img.shields.io/codecov/c/github/expert360/dal.svg)
 ![Hex.pm](https://img.shields.io/hexpm/dt/dal.svg)
 ![Hex.pm](https://img.shields.io/hexpm/v/dal.svg)
 [![Inline docs](http://inch-ci.org/github/expert360/dal.svg)](http://inch-ci.org/github/expert360/dal)
 
+Global Data Access Layer to replace direct use of `Ecto.Repo` and other data services.
 
-**TODO: Add description**
+The DAL performs two core functions:
+
+1. Provides an abstraction for other data repositories
+2. Provides a set of macros representing all available data types that can be used in services
```

Comment:
It doesn't and shouldn't do this. That was the very poorly named DALTypes which has been deprecated.

I can flesh out the README later